vm/src/stdlib/itertools.rs

```diff
@@ -1028,23 +1114,17 @@ mod decl {
         }
     }
 
-    #[derive(FromArgs)]
-    struct ProductArgs {
-        #[pyarg(named, optional)]
-        repeat: OptionalArg<usize>,
-    }
```

Comment:
to keep the original behaviour,
`type Args = (Args<PyObjectRef>, ProductArgs);`  will work.

Reply:
I would fix it! but putting iterable to Args which derive `FromArgs` cause Error due to Args does not implement `TryFromBorrowedObject` (maybe it does not have trait Sized?). To remain the original, I would use the first suggestion

Reply:
Not sure why it looked like related to TryFormBorrowedObject, but I think `vm::function::Args` can cause some problem.
The first approach will work.

```rust
    #[derive(FromArgs)]
    struct ProductArgs {
        #[pyarg(named, optional)]
        repeat: OptionalArg<usize>,
    }

    impl SlotConstructor for PyItertoolsProduct {
        type Args = (Args<PyObjectRef>, ProductArgs);
        fn py_new(cls: PyTypeRef, (iterables, args): Self::Args, vm: &VirtualMachine) -> PyResult {
            let repeat = args.repeat.unwrap_or(1);
            
```